src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.InternalBufferSize.cs

```diff
@@ -44,40 +44,80 @@ public class InternalBufferSizeTests : FileSystemWatcherTest
         [PlatformSpecific(TestPlatforms.Windows)]  // Uses P/Invokes
         public void FileSystemWatcher_InternalBufferSize(bool setToHigherCapacity)
         {
+            ManualResetEvent unblockHandler = new ManualResetEvent(false);
             using (var testDirectory = new TempDirectory(GetTestFilePath()))
             using (var file = new TempFile(Path.Combine(testDirectory.Path, "file")))
-            using (var watcher = new FileSystemWatcher(testDirectory.Path, Path.GetFileName(file.Path)))
+            using (FileSystemWatcher watcher = CreateWatcher(testDirectory.Path, file.Path, unblockHandler))
             {
-                int internalBufferOperationCapacity = watcher.InternalBufferSize / (17 + Path.GetFileName(file.Path).Length);
+                int internalBufferOperationCapacity = CalculateInternalBufferOperationCapacity(watcher.InternalBufferSize, file.Path);
 
                 // Set the capacity high to ensure no error events arise.
                 if (setToHigherCapacity)
                     watcher.InternalBufferSize = watcher.InternalBufferSize * 12;
 
-                // block the handling thread
-                ManualResetEvent unblockHandler = new ManualResetEvent(false);
-                watcher.Changed += (o, e) =>
-                {
-                    unblockHandler.WaitOne();
-                };
-
-                // generate enough file change events to overflow the default buffer
-                Action action = () =>
-                {
-                    for (int i = 1; i < internalBufferOperationCapacity * 10; i++)
-                    {
-                        File.SetLastWriteTime(file.Path, DateTime.Now + TimeSpan.FromSeconds(i));
-                    }
-
-                    unblockHandler.Set();
-                };
-                Action cleanup = () => unblockHandler.Reset();
+                Action action = GetAction(unblockHandler, internalBufferOperationCapacity, file.Path);
+                Action cleanup = GetCleanup(unblockHandler);
 
                 if (setToHigherCapacity)
                     ExpectNoError(watcher, action, cleanup);
                 else
                     ExpectError(watcher, action, cleanup);
             }
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]  // Uses P/Invokes
```

Comment:
To answer your question, no, I don't believe this comment is still valid.  I'm guessing it's marked Windows-specific because it's relying on implementation details around Windows buffering, but that's just a guess.

Reply:
I've removed this comment.